mixer/pkg/runtime/dispatcher/session.go

```diff
@@ -231,9 +232,8 @@ func (s *session) waitForDispatched() {
 			st = state.checkResult.Status
 
 		case tpb.TEMPLATE_VARIETY_QUOTA:
-			if s.quotaResult == nil {
-				r := state.quotaResult
-				s.quotaResult = &r
+			if status.IsOK(s.quotaResult.Status) && s.quotaResult.ValidDuration == 0 && s.quotaResult.Amount == 0 {
```

Comment:
Nit: add an IsDefault() / IsEmpty() method on QuotaResult?

Reply:
+1. This also strikes me as begging to define "IsEmpty" method here and for `CheckResult`.